spotinst/organization_user/fields_spotinst_organization_user.go

```diff
@@ -0,0 +1,329 @@
+			orgUserWrapper := resourceObject.(*commons.OrgUserWrapper)
+			orgUser := orgUserWrapper.GetOrgUser()
+			if v, ok := resourceData.GetOk(string(Email)); ok && v != "" {
+				orgUser.SetEmail(spotinst.String(resourceData.Get(string(Email)).(string)))
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgUserWrapper := resourceObject.(*commons.OrgUserWrapper)
+			orgUser := orgUserWrapper.GetOrgUser()
+			if v, ok := resourceData.GetOk(string(Email)); ok && v != "" {
+				orgUser.SetEmail(spotinst.String(resourceData.Get(string(Email)).(string)))
+			}
+			return nil
+		},
+		nil,
+	)
+
+	fieldsMap[FirstName] = commons.NewGenericField(
+		commons.OrganizationUser,
+		FirstName,
+		&schema.Schema{
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgUserWrapper := resourceObject.(*commons.OrgUserWrapper)
+			orgUser := orgUserWrapper.GetOrgUser()
+			if v, ok := resourceData.GetOk(string(FirstName)); ok && v != "" {
+				orgUser.SetFirstName(spotinst.String(resourceData.Get(string(FirstName)).(string)))
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgUserWrapper := resourceObject.(*commons.OrgUserWrapper)
+			orgUser := orgUserWrapper.GetOrgUser()
+			if v, ok := resourceData.GetOk(string(FirstName)); ok && v != "" {
+				orgUser.SetFirstName(spotinst.String(resourceData.Get(string(FirstName)).(string)))
+			}
+			return nil
+		},
+		nil,
+	)
+
+	fieldsMap[LastName] = commons.NewGenericField(
+		commons.OrganizationUser,
+		LastName,
+		&schema.Schema{
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgUserWrapper := resourceObject.(*commons.OrgUserWrapper)
+			orgUser := orgUserWrapper.GetOrgUser()
+			if v, ok := resourceData.GetOk(string(LastName)); ok && v != "" {
+				orgUser.SetLastName(spotinst.String(resourceData.Get(string(LastName)).(string)))
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			orgUserWrapper := resourceObject.(*commons.OrgUserWrapper)
+			orgUser := orgUserWrapper.GetOrgUser()
+			if v, ok := resourceData.GetOk(string(LastName)); ok && v != "" {
+				orgUser.SetLastName(spotinst.String(resourceData.Get(string(LastName)).(string)))
+			}
+			return nil
+		},
+		nil,
+	)
+
+	fieldsMap[Password] = commons.NewGenericField(
+		commons.OrganizationUser,
+		Password,
+		&schema.Schema{
+			Type:     schema.TypeString,
+			Optional: true,
```

Comment:
in the resource document you have mentioned it as required, please modify which is right

Reply:
it is optional i have modified to optional in example